src/databricks/labs/dqx/llm/README.md

```diff
@@ -0,0 +1,394 @@
+## 🤖 LLM-Assisted Features
+
+This module provides **optional** LLM-based primary key detection capabilities for the DQX data quality framework. The functionality is completely optional and only activates when users explicitly request it. Primary key detection can be used during data profiling and can also be enabled for `compare_datasets` checks to improve data comparison accuracy.
+
+## 🎯 **Overview**
+
+The LLM-based Primary Key Detection uses Large Language Models (via DSPy and Databricks Model Serving) to intelligently identify primary keys from table schema and metadata. This enhances the DQX profiling process by automatically detecting primary keys and generating appropriate uniqueness validation rules.
+
+## ✅ **Key Features**
+
+- **🔧 Completely Optional**: Not activated by default - requires explicit enablement
+- **🤖 Intelligent Detection**: Uses LLM analysis of table schema and metadata
+- **✨ Multiple Activation Methods**: Various ways to enable when needed
+- **🛡️ Graceful Fallback**: Clear messaging when dependencies unavailable
+- **⚡ Performance Optimized**: Lazy loading and conditional execution
+- **🔍 Duplicate Validation**: Optionally validates detected PKs for duplicates
+- **📊 Confidence Scoring**: Provides confidence levels and reasoning
+- **🔄 Retry Logic**: Handles cases where initial detection finds duplicates
+
+## 📦 **Installation**
+
+### **LLM-Enhanced Usage**
+```bash
+# Install DQX with LLM dependencies using extras
+databricks labs install dqx[llm]
+
+# Now you can enable LLM features when needed
+from databricks.labs.dqx.config import ProfilerConfig, LLMConfig
+config = ProfilerConfig(llm_config=LLMConfig(enable_pk_detection=True))
+```
+
+## 🚀 **Usage Examples**
+
+### **Method 1: Configuration-Based**
+```python
+from databricks.labs.dqx.config import ProfilerConfig, LLMConfig
+from databricks.labs.dqx.profiler.runner import ProfilerRunner
+
+# Default configuration (LLM features disabled)
+config = ProfilerConfig()
+
+# Enable LLM-based PK detection via configuration
+config = ProfilerConfig(
+    llm_config=LLMConfig(
+        enable_pk_detection=True,
+        pk_detection_endpoint="databricks-meta-llama-3-1-8b-instruct"
+    )
+)
+
+runner = ProfilerRunner(ws, spark, installation, profiler, generator)
```

Comment:
runner is meant to be used with workflows/jobs. Method 1 should show how to enable this pk detection with profiler job. Simply showing which configuration needs to be provided.